sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/Response/ConnectEventResponse.cs

```diff
@@ -1,45 +1,77 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
 using System.Collections.Generic;
+using System.ComponentModel;
 using System.Linq;
+using System.Runtime.Serialization;
 using System.Text.Json.Serialization;
 
 namespace Microsoft.Azure.WebPubSub.Common
 {
     /// <summary>
     /// Response for connect event.
     /// </summary>
+    [DataContract]
     public class ConnectEventResponse : WebPubSubEventResponse
     {
-        private Dictionary<string, object> _states = new();
+        private Dictionary<string, BinaryData> _states;
+
+        [DataMember(Name = "code")]
+        internal override WebPubSubStatusCode StatusCode
+        {
+            get
+            {
+                return WebPubSubStatusCode.Success;
+            }
+            set
+            {
+                if (value != WebPubSubStatusCode.Success)
+                {
+                    throw new ArgumentException("StatusCode shouldn't be set to errors in a normal connect event.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// The connection states.
+        /// </summary>
+        [IgnoreDataMember]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public IReadOnlyDictionary<string, object> States { get; private set; }
 
         /// <summary>
         /// The connection states.
         /// </summary>
-        public IReadOnlyDictionary<string, object> States => _states;
+        [IgnoreDataMember]
+        public IReadOnlyDictionary<string, BinaryData> ConnectionStates => _states;
```

Comment:
Why is this one ignored too? We don't serialize the states?

Reply:
`States` here doesn't applied to `System.Text.Json` deserialization logic(base64 related). And we have special logic to handle the states and merge with request.  See [HERE](https://github.com/tg-msft/azure-sdk-for-net/blob/wps-binary-data-states/sdk/webpubsub/Microsoft.Azure.WebJobs.Extensions.WebPubSub/src/Utilities.cs#L228) for the usage.